File: apps/desktop/Sources/Crisp/Models/Strength.swift
/// How aggressively to cut. Each preset maps to the engine's pause threshold and
/// the breathing room kept around every cut.
enum Strength: String, CaseIterable, Identifiable {
    case gentle = "Gentle"
    case balanced = "Balanced"
    case aggressive = "Aggressive"
    case veryAggressive = "Very aggressive"

    var id: String { rawValue }

    var detail: String {
        switch self {
        case .gentle:         return "Cuts only clearly long pauses. Most natural."
        case .balanced:       return "A safe middle ground."
        case .aggressive:     return "Cuts short \u{201C}thinking\u{201D} gaps too. Recommended."
        case .veryAggressive: return "Tightest possible. Can feel fast-paced."
        }
    }
    var pause: Double {
        switch self {
        case .gentle: return 0.80
        case .balanced: return 0.60
        case .aggressive: return 0.35
        case .veryAggressive: return 0.25
        }
    }
    var keepPause: Double {
        switch self {
        case .gentle: return 0.18
        case .balanced: return 0.15
        case .aggressive: return 0.10
        case .veryAggressive: return 0.08
        }
    }
}

File: apps/desktop-win/Models/Strength.cs
using System.Collections.Generic;

namespace Crisp.Models;

/// How aggressively to cut. Port of macOS CrispCore/Models/Strength.swift — each
/// preset maps to the engine's --pause threshold and the --keep-pause breathing
/// room kept around every cut. (Custom's real values live in settings; until the
/// Settings screen is ported it falls back to Aggressive, same as Swift.)
public enum Strength
{
    Gentle,
    Balanced,
    Aggressive,
    VeryAggressive,
    Custom,
}

public sealed record StrengthPreset(Strength Value, string Name, string PickerLabel, string Detail, double Pause, double KeepPause);

public static class Strengths
{
    public static readonly IReadOnlyList<StrengthPreset> All = new[]
    {
        new StrengthPreset(Strength.Gentle, "Gentle", "Gentle",
            "Cuts only clearly long pauses. Most natural.", 0.80, 0.18),
        new StrengthPreset(Strength.Balanced, "Balanced", "Balanced",
            "A safe middle ground.", 0.60, 0.15),
        new StrengthPreset(Strength.Aggressive, "Aggressive", "Aggressive",
            "Cuts short “thinking” gaps too. Recommended.", 0.35, 0.10),
        new StrengthPreset(Strength.VeryAggressive, "Very aggressive", "Very",
            "Tightest possible. Can feel fast-paced.", 0.25, 0.08),
        new StrengthPreset(Strength.Custom, "Custom", "Custom",
            "Your own settings — adjust them in Settings.", 0.35, 0.10),
    };

    public static StrengthPreset Of(Strength s)
    {
        foreach (var p in All) if (p.Value == s) return p;
        return All[2]; // Aggressive
    }

    /// The engine flags this strength implies (pauses-only knobs).
    public static IEnumerable<string> ToArgs(Strength s)
    {
        var p = Of(s);
        yield return "--pause"; yield return p.Pause.ToString(System.Globalization.CultureInfo.InvariantCulture);
        yield return "--keep-pause"; yield return p.KeepPause.ToString(System.Globalization.CultureInfo.InvariantCulture);
    }
}
